pkg/domain/cloud.go

```diff
@@ -12,7 +15,40 @@ type CloudAssetChanges struct {
 	AccountID    string
 	Region       string
 	ResourceID   string
-	Tags         map[string]string
+	ARN          string
+	Tags         InboundResourceTags
+}
+
+// InboundResourceTags represents the contents of the metadata (CloudAssetChanges.Tags) saved with the resource
+type InboundResourceTags map[string]interface{}
+
+// Value implements the driver.Valuer interface to marshal InboundResourceTags to []byte
+func (p InboundResourceTags) Value() (driver.Value, error) {
+	j, err := json.Marshal(p)
+	return j, err
+}
+
+// Scan implements the sql.Scanner interface to unmarshal the database data into a map, which is the same type as received and written into the database
+func (p *InboundResourceTags) Scan(src interface{}) error {
```

Comment:
I don't think this is the best place for this functionality. The intent of the domain package is to declare domain types that are used throughout the various packages in the project. This functions seems specific to the data layer only.